### PR TITLE
Add hyperref config

### PR DESCRIPTION
**Pull request by jonasrichardrichter**
This config replaces the boxes with coloring the link text. This change makes the document more readable. 

Applies only to the simple mode.

### CODE REVIEW

**Comment by Nuckal777** [comment]
The colored links would appear in a printout as well. Can you wrap this as an boolkey option for the basimple environment, like it is with the blocknotice? Should not be to hard. https://github.com/Nuckal777/ba-latex-template/blob/rewrite/baarticle.cls#L296
```latex
   \define@boolkey[BA]{simple}{blocknotice}[true]{}

   \presetkeys[BA]{simple}{type=undefined,blocknotice=true}{}
   ...
    \newenvironment{basimple}[1][]{
        \setkeys[BA]{simple}{#1}
       ...
        \ifBA@simple@blocknotice
            \mkblocknotice{
                signature=\@basimplesignature,
                date=\@basimplereturndate,
                location=\@basimplelocation
            }
        \fi
        ...
    }
```

**Comment by jonasrichardrichter** [comment]
Done ✅

The default value is false. Perhaps a note in the documentation would be handy. Maybe next week I have time for it.

File: baarticle.cls [hunk 1]
```diff
@@ -51,13 +51,15 @@
...
-    }
+    \ifBA@simple@linkcoloring
+        % hyperref config
+        \hypersetup{
+            colorlinks=true,
+            linkcolor=blue,
+            filecolor=magenta,      
+            urlcolor=cyan,
+        }
+     \fi
```
Comment (Nuckal777) [inline baarticle.cls:62]: This should very likely be placed within the definition basimple after the call to `\setkeys`. I feel above `\mktitle` is a good place (https://github.com/Nuckal777/ba-latex-template/blob/rewrite/baarticle.cls#L308). I would expect that the check after the imports is either always true or false right now (but I haven't tested).

File: baarticle.cls [hunk 2]
```diff
@@ -302,8 +304,10 @@
     \define@key[BA]{simple}{assignment}{\def\@basimpleassignment{#1}}
     % whether to include the blocknotice or not
     \define@boolkey[BA]{simple}{blocknotice}[true]{}
+    % display links with boxes (do not appear while printing) or as colorings (appear while printing)
```
Comment (Nuckal777) [inline baarticle.cls:307]: As a sidenote: These boxes are usally displayed by certain PDF readers and are not part of the actual PDF file.

**Review by Nuckal777 — COMMENTED** [review]
Thanks for your efforts. :+1: Feel free to add a note in the corresponding markdown file. https://github.com/Nuckal777/ba-latex-template/blob/rewrite/docs/usage/simple.md

**Pull request merged**